src/App.js

```diff
@@ -0,0 +1,77 @@
+import './App.css';
+import React from 'react'
+import { useState, useRef } from 'react';
+
+function App() {
+  const [res, setRes] = useState([]);
+  const [result, setResult] = useState([]);
```

Comment:
This was the only thing that caused me some confusion in this file. Could you potentially rename res and result to make their purposes more clear? Or add a comment here saying what each will do? I had a hard time figuring out what they did since their names were so similar.